docs/websites/varnish/use-varnish-with-nginx-to-serve-cached-wordpress-content-over-ssl-on-debian-8.md

```diff
@@ -0,0 +1,683 @@
+
+		sub vcl_synth {
+		 if (resp.status == 850) {  
+			 set resp.http.Location = req.http.x-redir;       
+			 set resp.status = 302;
+			 return (deliver);
+		 }
+		}
+
+		sub vcl_purge {
+		 set req.method = "GET";
+		 set req.http.X-Purger = "Purged";
+		 return (restart);
+		}
+
+		sub vcl_backend_response {
+		 set beresp.ttl = 24h;
+		 set beresp.grace = 1h;
+
+		 if (bereq.url !~ "wp-admin|wp-login|product|cart|checkout|my-account|/?remove_item=") {
+		 unset beresp.http.set-cookie;    
+		 }
+		}
+
+		sub vcl_deliver {
+		 if (req.http.X-Purger) {
+			set resp.http.X-Purger = req.http.X-Purger;
+		 }
+		}
+		~~~
+
+
+	For Varnish to work we also need to edit the `varnish.service` file to use our custom configuration file, with the port number and allocated memory values updated to match the changes we made in our `/etc/default/varnish` file.
+
+21. Open the `/lib/systemd/system/varnish.service` file:
+
+        sudo nano /lib/systemd/system/varnish.service
+
+22. Find the two lines beginning with  `ExecStart` and edit them to look like this:
+
+	{: .file-excerpt }
+	lib/systemd/system/varnish.service
+	:   ~~~ conf
+	
+		ExecStartPre=/usr/sbin/varnishd -C -f /etc/varnish/custom.vcl
+		ExecStart=/usr/sbin/varnishd -a :80 -T localhost:6082 -f /etc/varnish/custom.vcl -S /etc/varnish/secret -s malloc,500M
+		~~~
+
+23. After saving and exiting the file you have to reload the systemd process:
+
+        sudo systemctl daemon-reload
+
+24. To make Varnish start at boot up, just run the following command:
+
+		sudo systemctl enable varnish.service
+
+We won't start Varnish at this point because we need to configure nginx first.
+
+## Install and Configure PHP
+
+Before configuring nginx we have to install PHP-FPM. FPM is short for FastCGI Process Manager, and it allows the web server to act as a proxy passing all requests with the `.php` file extension to the PHP interpreter.
+
+1. To install PHP-FPM just run:
+
+        sudo apt-get install php5-fpm php5-mysql
+
+2. Open the `/etc/php5/fpm/php.ini` file to configure PHP-FPM:
+
+		sudo nano /etc/php5/fpm/php.ini
+
+3. Find the directive `cgi.fix_pathinfo=` and set it to `0`, for safety reasons. If this parameter is set to `1`, the PHP interpreter will try to process the file who's path is as close to the requested path as possible, if it's set to `0`, the interpreter will only process the file with the exact path, which is a safer alternative.
+
+		cgi.fix_pathinfo=0
+
+4. Now open `www.conf`:
+
+		sudo nano /etc/php5/fpm/pool.d/www.conf
+
+5. Edit the `listen =` directive to specify the socket that will be used by nginx to pass requests to PHP-FPM:
+
+		listen = /var/run/php5-fpm.sock
+
+6.  Restart PHP-FPM:
+
+		sudo systemctl restart php5-fpm
+
+7.  Now open `/etc/nginx/fastcgi_params`:
+
+		sudo nano /etc/nginx/fastcgi_params
+
+8.  Find the line `fastcgi_param  HTTPS  $https if_not_empty;`. Below it add the following two lines, necessary for nginx to interact with the FastCGI service:
```

Comment:
My default installation does not include the `fastcgi_param HTTPS $https if_not_empty;` directive in this file. Should we add it?